plugins/catalog-backend/src/ingestion/types.ts

```diff
@@ -68,3 +69,69 @@ export type ReadLocationError = {
   location: LocationSpec;
   error: Error;
 };
+
+//
+// LocationAnalyzer
+//
+
+export type LocationAnalyzer = {
+  generateConfig(
```

Comment:
Would probably call this `analyzeLocation` instead?

Reply:
Renamed.